drivers/include/ad7746.h

```diff
@@ -0,0 +1,279 @@
+    ad7746_vt_mode_t vt_mode;                 /**< mode of the voltage/temp ch */
+} ad7746_params_t;
+
+
+/**
+ * @brief   AD7746 device descriptor
+ */
+typedef struct ad7746 {
+    ad7746_params_t params; /**< device driver configuration */
+} ad7746_t;
+
+/**
+ * @brief   Initializes an AD7746 device
+ *
+ * @param[in,out] dev  device descriptor
+ * @param[in] params   device configuration
+ *
+ * @return AD7746_OK on success
+ * @return AD7746_NODEV if no device is found on the bus
+ * @return AD7746_NOI2C if other error occurs
+ */
+int ad7746_init(ad7746_t *dev, const ad7746_params_t *params);
+
+/**
+ * @brief   Reads a raw value either from the capacitance channel or from the
+ *          voltage / temperature channel if available
+ *
+ * @param[in] dev   device descriptor
+ * @param[in] ch    channel to read:
+ *                      - AD7746_READ_CAP_CH: Capacitance
+ *                      - AD7746_READ_VT_CH: Voltage / temperature
+ * @param[out] raw  read value
+ *
+ * @return AD7746_OK on success
+ * @return AD7746_NODATA if no data is available
+ * @return AD7746_I2C on error getting a reponse
+ */
+int ad7746_read_raw_ch(const ad7746_t *dev, uint8_t ch, uint32_t *raw);
+
+/**
+ * @brief   Convenience function that reads a raw value from the capacitance
+ *          channel
+ *
+ * @param[in] dev   device descriptor
+ * @param[out] raw  read value
+ *
+ * @return AD7746_OK on success
+ * @return AD7746_NODATA if no data is available
+ * @return AD7746_I2C on error getting a reponse
+ */
+static inline int ad7746_read_raw_cap_ch(const ad7746_t *dev, uint32_t *raw)
+{
+    return ad7746_read_raw_ch(dev, AD7746_READ_CAP_CH, raw);
+}
+
+/**
+ * @brief   Convenience function that reads a raw value from the voltage /
+ *          temperature channel
+ *
+ * @param[in] dev   device descriptor
+ * @param[out] raw  read value
+ *
+ * @return AD7746_OK on success
+ * @return AD7746_NODATA if no data is available
+ * @return AD7746_I2C on error getting a reponse
+ */
+static inline int ad7746_read_raw_vt_ch(const ad7746_t *dev, uint32_t *raw)
+{
+    return ad7746_read_raw_ch(dev, AD7746_READ_VT_CH, raw);
+}
+
+/**
+ * @brief   Converts a raw code into a capacitance expressed in fF
+ *
+ * @param[in] raw   Raw code from the device
+ *
+ * @return capacitance in fF
+ */
+int ad7746_raw_to_capacitance(uint32_t raw);
+
+/**
+ * @brief   Converts a raw code into temperature expressed in celsius (C)
+ *
+ * @param[in] raw   Raw code from the device
+ *
+ * @return temperature in celsius
+ */
+int ad7746_raw_to_temperature(uint32_t raw);
+
+/**
+ * @brief   Converts a raw code into voltage expressed in mV
+ *
+ * @param[in] raw   Raw code from the device
+ * @return voltage in mV
+ *
+ * @note Note when using the @ref AD7746_VT_MD_VDD mode, that the voltage from the
+ *       VDD pin is internally attenuated by 6
+ */
+int ad7746_raw_to_voltage(uint32_t raw);
+
+/**
+ * @brief   Sets the current input for the capacitive measurement
```

Comment:
Is this something one would change during runtime? If not, maybe we could get rid of the function and make it only compile-time configurable?

Reply:
The capacitance channel has two inputs (CIN1 and CIN2), this can be changed during runtime that's why this function exists. By default the selected one is CIN1, I can add that into the documentation.